src/planetary_system.jl

```diff
@@ -352,6 +352,340 @@ function generate_period_and_sizes_power_law(s::Star, sim_param::SimParam; num_p
 end
 
 
+
+
+
+using SpecialFunctions
+
+cdf_lognormal(x::Float64; μ::Float64=0., σ::Float64=1.) = 0.5*erfc(-(log(x) - μ)/(σ*sqrt(2.)))
+
+function invert_cdf_lognormal(y::Float64; μ::Float64=0., σ::Float64=1.)
+    @assert 0<=y<=1
+    return exp(μ - sqrt(2.)*σ*erfcinv(2*y))
+end
+
+function cdf_power_law(x::Float64, x0::Float64, x1::Float64, α::Float64)
+    @assert x0 <= x <= x1
+    if α != -1
+        return (x^(α+1) - x0^(α+1))/(x1^(α+1) - x0^(α+1))
+    else # if α == -1
+        return log(x/x0)/log(x1/x0)
+    end
+end
+
+function invert_cdf_power_law(y::Float64, x0::Float64, x1::Float64, α::Float64)
+    @assert x0 < x1
+    if α != -1
+        return (x0^(α+1) + (x1^(α+1) - x0^(α+1))*y)^(1/(α+1))
+    else # if α == -1
+        return x0*(x1/x0)^y
+    end
+end
+
+function draw_segmented_uniform(segments::Array{Tuple{Float64,Float64},1})
+    # Note: "segments" must be non-overlapping
+    n_segs = length(segments) # number of segments
+    if n_segs == 0
+        println("No segments given; returning NaN.")
```

Comment:
Is this supposed to happen?  Or does this mean there's a problem.  Would @warn or @error be more appropriate?

Reply:
Yeah, the idea is that this should never happen. I've changed it to "@assert n_segs > 0" now...

Reply:
Thanks.